src/routes/article.ts

```diff
@@ -0,0 +1,35 @@
+import createHttpError from 'http-errors';
+import { OK } from 'http-status-codes';
+import {
+  DefaultStateExtends, Middleware, Next,
+} from 'koa';
+import url from 'url';
+import { AppContext } from '../app';
+import { namedNode } from '../data-factory';
+import ArticleNotFound from '../errors/article-not-found';
+
+export default (): Middleware<DefaultStateExtends, AppContext> => (
+  async ({
+    path, articles, request, response,
+  }: AppContext, next: Next): Promise<void> => {
+    try {
+      await next();
+    } catch (error) {
+      if (!(error instanceof createHttpError.NotFound)) {
+        throw error;
+      }
+
+      try {
```

Comment:
If you add a `return` in the end of the `try` and close the `catch` straight after the `if` statement, we can avoid nesting `try` blocks.